### ISSUE TITLE
[ET:Legacy mod] Objectives not shown on map

### ISSUE THREAD

**Issue by rorgoroth** [etfdevs/ETe#71]
It works with local server as etmain.

But playing online, the server running legacy 2.78 none of the objectives (or spawn flags) show.
It's all there and you can hover over it, it's like the icons are stuck under the map because the text is on top as it should be.

![a](https://user-images.githubusercontent.com/77244135/138441604-f5e00e86-a076-4565-ac93-99b5859a1531.png)
![b](https://user-images.githubusercontent.com/77244135/138441607-ee2d21fb-7a7e-49ec-a609-117c300dfde6.png)

**Comment by ryzyk-krzysiek** [comment]
Hello, the bug happens with limbo map, command map (`G`) and square minimap (`cg_drawcompass 2`), the common thing among them is that they use different shader `cgs.media.blackmask`:

https://github.com/etlegacy/etlegacy/blob/b415dfb69598c40149c22af5ff9d2a1cae7fda16/src/cgame/cg_main.c#L1735

Shader:

https://github.com/etlegacy/etlegacy/blob/e7922f1d1d2d7918c818bc9a40e5f9bb7609156e/etmain/scripts/legacy.shader#L3-L15

After deleting `blendfunc blend` the issue is gone in ETe and in ETL everything seems to work fine (I haven't play tested every map though). I have no real knowledge how shaders work but maybe this will help you track the potential issue in ETe, or change the shader in ETL if this `blendfunc` is incorrectly used or not needed.

![obraz](https://user-images.githubusercontent.com/4499367/162640479-6177bc36-b7b6-4afa-b52f-3f2d446af938.png)

**Comment by ensiform** [comment]
What is the purpose behind the different blending over the original?

**Comment by ryzyk-krzysiek** [comment]
> What is the purpose behind the different blending over the original?

No idea really, only related information I found is this issue https://github.com/etlegacy/etlegacy/issues/251 but there isn't anything really, and this comment:

`cgs.media.blackmask                         = trap_R_RegisterShaderNoMip("images/blackmask"); // etpro icons support`

ETPro shader:

```
etpro/images/blackmask
{
	nopicmip 
	nocompress
	nomipmaps
	{
		map gfx/2d/mapmask.tga
		depthwrite
		blendfunc blend
		rgbGen identity
		alphaGen vertex
	}
}
```

So it looks like ETL uses exactly the same shader as is used in ETPro which uses vanilla 2.60b client and it works there fine too.

**Comment by ensiform** [comment]
This is fixed in 8c27d93c407dcfaed46a7e55464647b3ac9e715c and that was made irrelevant by the overall fix in c4df15b020f9d38f78da05cbbac0dfdcdf42d174
